### PR TITLE
Support dynamic import call

### LINKED ISSUE #1728: Dynamic import syntax support
I couldn't find an issue tracking this already, but let me know if it is a duplicate. This may well be implementable now that it is Stage 3 (https://github.com/estree/estree/pull/137).

### PR DESCRIPTION
**Pull request by ariya**
Fix #1728

### CODE REVIEW

File: src/parser.ts [hunk 1]
```diff
@@ -1789,10 +1813,14 @@ export class Parser {
                     statement = this.parseExportDeclaration();
                     break;
                 case 'import':
-                    if (!this.context.isModule) {
-                        this.tolerateUnexpectedToken(this.lookahead, Messages.IllegalImportDeclaration);
+                    if (this.matchImportCall()) {
```
Comment (mikesherov) [inline src/parser.ts]: Oh, interesting, import can be called from scripts? Nice to know!

**Comment by ariya** [comment]
> Maybe a test showing it can't be used in a newExpression?

Good point, I can definitely at that.

File: src/parser.ts (hunk 1 above)
Comment (ariya) [inline src/parser.ts]: That's my interpretation, not sure if it's correct or not. @domenic do you have an opinion?

Comment (domenic) [inline src/parser.ts]: Yes, definitely.

**Comment by ariya** [comment]
Ah, I also still forget to tackle the case such as `await import(x)`.

**Comment by doplumi** [comment]
I get an error with a StackTrace that leads to

```
    parseImportDeclaration(): Node.ImportDeclaration {
        if (this.context.inFunctionBody) {
            this.throwError(Messages.IllegalImportDeclaration);
        }
```

in [parser.ts](https://github.com/jquery/esprima/blob/master/src/parser.ts#L3470-L3472), while parsing valid es2018.

It seems like `import` statements inside of functions are rejected.

**Comment by doplumi** [comment]
I get an error with a stack trace that leads to

```
parseImportDeclaration(): Node.ImportDeclaration {
    if (this.context.inFunctionBody) {
        this.throwError(Messages.IllegalImportDeclaration);
    }
```

in [parser.ts](https://github.com/jquery/esprima/blob/master/src/parser.ts#L3470-L3472), while parsing valid es2018.

It seems that `import` statements inside of functions are rejected.

**Pull request closed without merging**